src/feed/views/feed_view.py

```diff
@@ -169,3 +176,78 @@ def get_queryset(self):
             )
 
         return queryset.distinct()
+
+    def get_queryset_for_personalized(self, item_ids):
+        """
+        Get FeedEntry queryset for personalized recommendations.
+
+        """
+        if not item_ids:
+            return FeedEntry.objects.none()
+
+        item_ids = [int(item_id) for item_id in item_ids]
+
+        preserved_order = [
+            When(unified_document_id=item_id, then=Value(idx))
+            for idx, item_id in enumerate(item_ids)
+        ]
+
+        queryset = (
+            FeedEntry.objects.filter(unified_document_id__in=item_ids)
+            .order_by(Case(*preserved_order, output_field=IntegerField()))
+            .select_related(
+                "content_type",
+                "user",
+                "user__author_profile",
+                "user__userverification",
+            )
+            .distinct()
+        )
+
+        return queryset
+
+    @action(detail=False, methods=["get"], permission_classes=[IsAuthenticated])
+    def personalized(self, request):
+        """
+        Get personalized feed entries for the authenticated user using AWS Personalize.
+        """
+        try:
+            # Initialize Personalize client
+            personalize_client = PersonalizeClient()
+
+            page_size = int(
+                request.query_params.get("page_size", self.pagination_class.page_size)
+            )
+            # Request more items than page size to account for items that might not exist
+            num_results = min(page_size * 3, 100)
+
+            item_ids = personalize_client.get_recommendations_for_user(
+                user_id=request.user.id,
+                filter=request.query_params.get("filter"),
+                num_results=num_results,
+            )
+
+            if not item_ids:
+                logger.info(f"No recommendations returned for user {request.user.id}")
+                return Response(
+                    {"count": 0, "next": None, "previous": None, "results": []}
+                )
```

Comment:
Let the framework handle the serialization to get the expected share of an empty page, i.e., remove the `return` statement here.